diff -Nurb o.hk_classes-0.8.2/acinclude.m4 n.hk_classes-0.8.2/acinclude.m4
--- o.hk_classes-0.8.2/acinclude.m4	2006-10-03 12:33:07.000000000 +0200
+++ n.hk_classes-0.8.2/acinclude.m4	2006-11-19 11:16:41.000000000 +0100
@@ -57,13 +57,24 @@
 HK_CLASSESINCDIR=$with_hk_classes_incdir
 fi
 
+if test "$with_hk_classes_dir" != ""; then
+	HK_CLASSESDIR=$with_hk_classes_dir
+fi
+if test "$with_hk_classes_drvdir" != ""; then
+	HK_DRIVERSDIR=$with_hk_classes_drvdir
+else
+	HK_DRIVERSDIR=$HK_CLASSESDIR/drivers
+fi
+AC_SUBST(HK_DRIVERSDIR)
+
 HKPYTHONDIR=""
 UTILITIESDIR=""
 AC_SUBST(HK_CLASSESDIR)
 AC_SUBST(HK_CLASSESINCDIR)
 NODRIVER=true
 POSSIBLE_DRIVERS="FIREBIRD,MYSQL,ODBC,PARADOX,POSTGRESQL,SQLITE2,SQLITE3,MDB,XBASE"
-dnl AC_DEFINE_UNQUOTED(HKCLASSES, "$HK_CLASSESDIR",[PREPROCESSOR DEFINITION])
+AC_DEFINE_UNQUOTED(HKCLASSES, "$HK_CLASSESDIR", [HKCLASSES - directory for libhk_classes library])
+AC_DEFINE_UNQUOTED(HK_DRIVERSDIR, "$HK_DRIVERSDIR", [HK_DRIVERSDIR - database drivers' directory])
 AC_SUBST(BITARCHITECTURE)
 ])
 
@@ -437,6 +448,7 @@
   dnl accordingly.
   dnl AC_LIB_LINKFLAGS_BODY([iconv])
 
+  AC_CHECK_FUNC(iconv, [ac_liconv="no"], [
   AC_MSG_CHECKING([if libiconv is present])
   ac_save_LIBS="$LIBS"
   LIBS="$ac_save_LIBS -liconv"
@@ -447,6 +459,7 @@
     LIBS="$ac_save_LIBS"
     ac_liconv="no")
   AC_MSG_RESULT($ac_liconv)
+  ])
   dnl Add $INCICONV to CPPFLAGS before performing the following checks,
   dnl because if the user has installed libiconv and not disabled its use
   dnl via --without-libiconv-prefix, he wants to use it. The first
diff -Nurb o.hk_classes-0.8.2/configure.in n.hk_classes-0.8.2/configure.in
--- o.hk_classes-0.8.2/configure.in	2006-10-23 17:33:56.000000000 +0200
+++ n.hk_classes-0.8.2/configure.in	2006-11-19 12:40:47.000000000 +0100
@@ -18,6 +18,8 @@
 AC_ARG_WITH(mysql-incdir,[  --with-mysql-incdir=PATH   Path to the Mysql header files],:,:)
 AC_ARG_WITH(mysql-libdir,[  --with-mysql-libdir=PATH   Path to the Mysql library],:,:)
 AC_ARG_WITH(hk_classes-incdir,[  --with-hk_classes-incdir=PATH   Path for the hk_classes header files],:,:)
+AC_ARG_WITH(hk_classes-dir,[  --with-hk_classes-dir=PATH   Path for the hk_classes library files],:,:)
+AC_ARG_WITH(hk_classes-drvdir,[  --with-hk_classes-drvdir=PATH   Path for the hk_classes driver files],:,:)
 
 AC_ARG_WITH(postgres, AC_HELP_STRING([--without-postgres],[disable postgres plugins]),[want_postgres=$withval], want_postgres=yes)
 AC_ARG_WITH(postgres-dir,[  --with-postgres-dir=PATH   Path to the postgres directory],:,:)
@@ -84,7 +86,7 @@
 HK_FIREBIRD2TEST
 HK_MDBTEST
 HK_MYSQLTEST
-PXLIB_BUILD_DATE=`date '+%Y-%m-%d'`
+PXLIB_BUILD_DATE='9999-12-31'
 AC_DEFINE_UNQUOTED(PXLIB_BUILD_DATE,"$PXLIB_BUILD_DATE",[pxlib build date])
 AC_SUBST(PXLIB_BUILD_DATE)
 
diff -Nurb o.hk_classes-0.8.2/hk_classes/hk_drivermanager.cpp n.hk_classes-0.8.2/hk_classes/hk_drivermanager.cpp
--- o.hk_classes-0.8.2/hk_classes/hk_drivermanager.cpp	2006-10-03 12:33:06.000000000 +0200
+++ n.hk_classes-0.8.2/hk_classes/hk_drivermanager.cpp	2006-11-19 13:06:54.000000000 +0100
@@ -59,7 +59,7 @@
 };
 
 requestdialogtype* hk_drivermanager::p_driverselectdialog=NULL;
-hk_string hk_drivermanager::p_hk_classespath=(hk_string)HKCLASSES+"/drivers";
+hk_string hk_drivermanager::p_hk_classespath=(hk_string)HK_DRIVERSDIR;
 vector<hk_string> hk_drivermanager::p_driverlist;
 
 typedef hk_connection*  connectiontype(hk_drivermanager* );
@@ -82,8 +82,7 @@
 {
     p_private = new hk_drivermanagerprivate;
     hk_report::setup_reportbasics();
-    p_hk_classespath=HKCLASSES;
-    p_hk_classespath=p_hk_classespath+"/drivers";
+    p_hk_classespath=HK_DRIVERSDIR;
     scan_directory();
     char* h= getenv("HOME");
     hk_string classespath=(h==NULL?"/tmp":h);
@@ -447,8 +446,7 @@
             #endif
             if (b.find("/usr/local/hk_classes")!=hk_string::npos)
             {
-                b=HKCLASSES;b+="/drivers";
-
+                b=HK_DRIVERSDIR;
                 set_path(b);
                 show_warningmessage("Due to a new directory hierarchy, your preferences have been changed");
                 save_preferences();
diff -Nurb o.hk_classes-0.8.2/hk_classes/hk_drivermanager.h n.hk_classes-0.8.2/hk_classes/hk_drivermanager.h
--- o.hk_classes-0.8.2/hk_classes/hk_drivermanager.h	2006-05-20 22:07:18.000000000 +0200
+++ n.hk_classes-0.8.2/hk_classes/hk_drivermanager.h	2006-11-19 12:49:56.000000000 +0100
@@ -17,6 +17,14 @@
 #include "hk_url.h"
 #include "hk_definitions.h"
 #include "hk_string.h"
+
+#ifndef HK_DRIVERSDIR
+#ifndef HKCLASSES
+#define HKCLASSES "/usr/local/lib/hk_classes"
+#endif
+#define HK_DRIVERSDIR HKCLASSES "/driver"
+#endif
+
 using namespace std;
 
 class hk_connection;
diff -Nurb o.hk_classes-0.8.2/hk_dbaseclasses/Makefile.am n.hk_classes-0.8.2/hk_dbaseclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_dbaseclasses/Makefile.am	2006-10-02 19:54:17.000000000 +0200
+++ n.hk_classes-0.8.2/hk_dbaseclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes 
 
 AM_CPPFLAGS=
diff -Nurb o.hk_classes-0.8.2/hk_firebirdclasses/Makefile.am n.hk_classes-0.8.2/hk_firebirdclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_firebirdclasses/Makefile.am	2006-10-02 19:54:17.000000000 +0200
+++ n.hk_classes-0.8.2/hk_firebirdclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@FIREBIRDINCDIR@
 
 AM_CPPFLAGS=-W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings
diff -Nurb o.hk_classes-0.8.2/hk_mdbclasses/Makefile.am n.hk_classes-0.8.2/hk_mdbclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_mdbclasses/Makefile.am	2006-10-02 19:54:17.000000000 +0200
+++ n.hk_classes-0.8.2/hk_mdbclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I ./mdbtools 
 
 AM_CPPFLAGS=$(GLIB_CFLAGS) 
diff -Nurb o.hk_classes-0.8.2/hk_mysqlclasses/Makefile.am n.hk_classes-0.8.2/hk_mysqlclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_mysqlclasses/Makefile.am	2006-10-02 19:54:17.000000000 +0200
+++ n.hk_classes-0.8.2/hk_mysqlclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@MYSQLINCDIR@
 
 AM_CXXFLAGS= -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I@XMLINCLUDE@ @BITARCHITECTURE@
diff -Nurb o.hk_classes-0.8.2/hk_odbcclasses/Makefile.am n.hk_classes-0.8.2/hk_odbcclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_odbcclasses/Makefile.am	2006-10-02 19:54:17.000000000 +0200
+++ n.hk_classes-0.8.2/hk_odbcclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@ODBCINCDIR@ 
 AM_CXXFLAGS= -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I@XMLINCLUDE@ @BITARCHITECTURE@
 
diff -Nurb o.hk_classes-0.8.2/hk_paradoxclasses/Makefile.am n.hk_classes-0.8.2/hk_paradoxclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_paradoxclasses/Makefile.am	2006-10-02 19:54:20.000000000 +0200
+++ n.hk_classes-0.8.2/hk_paradoxclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I./pxlib
 
 AM_CPPFLAGS=
diff -Nurb o.hk_classes-0.8.2/hk_postgresclasses/Makefile.am n.hk_classes-0.8.2/hk_postgresclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_postgresclasses/Makefile.am	2006-10-02 19:54:20.000000000 +0200
+++ n.hk_classes-0.8.2/hk_postgresclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES=   -I../hk_classes -I@POSTGRESINCDIR@
 
 AM_CXXFLAGS= -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I@XMLINCLUDE@ @BITARCHITECTURE@
diff -Nurb o.hk_classes-0.8.2/hk_sqlite3classes/Makefile.am n.hk_classes-0.8.2/hk_sqlite3classes/Makefile.am
--- o.hk_classes-0.8.2/hk_sqlite3classes/Makefile.am	2006-10-02 19:54:20.000000000 +0200
+++ n.hk_classes-0.8.2/hk_sqlite3classes/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,5 +1,5 @@
 
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 
 AM_CXXFLAGS= -Wall -DNO_TCL=1 -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I@XMLINCLUDE@ @BITARCHITECTURE@
 AM_CFLAGS= -DNO_TCL=1
diff -Nurb o.hk_classes-0.8.2/hk_sqliteclasses/Makefile.am n.hk_classes-0.8.2/hk_sqliteclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_sqliteclasses/Makefile.am	2006-10-02 19:54:20.000000000 +0200
+++ n.hk_classes-0.8.2/hk_sqliteclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 
 AM_CXXFLAGS= -Wall -DNO_TCL=1  -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I@XMLINCLUDE@ @BITARCHITECTURE@
 AM_CFLAGS= -DNO_TCL=1
diff -Nurb o.hk_classes-0.8.2/hk_xbaseclasses/Makefile.am n.hk_classes-0.8.2/hk_xbaseclasses/Makefile.am
--- o.hk_classes-0.8.2/hk_xbaseclasses/Makefile.am	2006-10-02 19:54:20.000000000 +0200
+++ n.hk_classes-0.8.2/hk_xbaseclasses/Makefile.am	2006-11-19 13:06:00.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@XBASEINCDIR@ 
 
 AM_CPPFLAGS=
