--- hk_classes-0.5.5/acinclude.m4.orig	Sun Aug 18 16:35:37 2002
+++ hk_classes-0.5.5/acinclude.m4	Thu Nov 21 11:47:35 2002
@@ -29,6 +29,15 @@
  HK_CLASSESDIR=$prefix
  HK_CLASSESINCDIR=$HK_CLASSESDIR/include
 fi
+if test "$with_hk_classes_dir" != ""; then
+HK_CLASSESDIR=$with_hk_classes_dir
+fi
+if test "$with_hk_classes_drvdir" != ""; then
+HK_DRIVERSDIR=$with_hk_classes_drvdir
+else
+HK_DRIVERSDIR=$HK_CLASSESDIR/drivers
+fi
+AC_SUBST(HK_DRIVERSDIR)
 if test "$with_hk_classes_incdir" != ""; then
 HK_CLASSESINCDIR=$with_hk_classes_incdir
 fi
--- hk_classes-0.5.5/configure.in.orig	Sat Sep 21 16:41:31 2002
+++ hk_classes-0.5.5/configure.in	Thu Nov 21 11:48:16 2002
@@ -8,6 +8,8 @@
 AC_ARG_WITH(mysql-dir,[  --with-mysql-dir=PATH   Path to the Mysql directory],:,:)
 AC_ARG_WITH(mysql-incdir,[  --with-mysql-incdir=PATH   Path to the Mysql header files],:,:)
 AC_ARG_WITH(mysql-libdir,[  --with-mysql-libdir=PATH   Path to the Mysql library],:,:)
+AC_ARG_WITH(hk_classes-dir,[  --with-hk_classes-dir=PATH   Path for the hk_classes library files],:,:)
+AC_ARG_WITH(hk_classes-drvdir,[  --with-hk_classes-drvdir=PATH   Path for the hk_classes driver files],:,:)
 AC_ARG_WITH(hk_classes-incdir,[  --with-hk_classes-incdir=PATH   Path for the hk_classes header files],:,:)
 
 AC_ARG_WITH(postgres-dir,[  --with-postgres-dir=PATH   Path to the postgres directory],:,:)
--- hk_classes-0.5.5/hk_mysqlclasses/Makefile.am.orig	Wed Sep  4 19:11:59 2002
+++ hk_classes-0.5.5/hk_mysqlclasses/Makefile.am	Thu Nov 21 12:44:43 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@MYSQLINCDIR@
 AM_CXXFLAGS= -Wall
 
--- hk_classes-0.5.5/hk_postgresclasses/Makefile.am.orig	Tue Aug 13 19:58:33 2002
+++ hk_classes-0.5.5/hk_postgresclasses/Makefile.am	Thu Nov 21 12:44:58 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES=   -I../hk_classes -I@POSTGRESINCDIR@
 
 libhk_postgresdriver_la_LDFLAGS = -version-info 0:0:0 -lpq  -L../hk_classes -L@POSTGRESLIBDIR@
--- hk_classes-0.5.5/hk_odbcclasses/Makefile.am.orig	Mon Nov  4 19:41:17 2002
+++ hk_classes-0.5.5/hk_odbcclasses/Makefile.am	Tue Dec  3 17:13:54 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@ODBCINCDIR@ 
 AM_CXXFLAGS= -Wall
 
--- hk_classes-0.7.2/hk_sqliteclasses/Makefile.am.orig	2005-02-18 08:47:12.000000000 +0100
+++ hk_classes-0.7.2/hk_sqliteclasses/Makefile.am	2005-02-18 08:49:37.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@SQLITEINCDIR@
 
 AM_CXXFLAGS= -Wall
--- hk_classes-0.7.2/hk_sqlite3classes/Makefile.am.orig	2005-02-18 08:49:58.000000000 +0100
+++ hk_classes-0.7.2/hk_sqlite3classes/Makefile.am	2005-02-18 08:50:08.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@SQLITE3INCDIR@
 
 AM_CXXFLAGS= -Wall
--- hk_classes-0.7.4/hk_firebirdclasses/Makefile.am~	2005-07-09 22:06:23.154547992 +0200
+++ hk_classes-0.7.4/hk_firebirdclasses/Makefile.am	2005-07-09 22:05:32.071313824 +0200
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@FIREBIRDINCDIR@
 
 AM_CPPFLAGS=
--- hk_classes-0.8/hk_xbaseclasses/Makefile.am~	2005-10-31 21:44:07.703534256 +0100
+++ hk_classes-0.8/hk_xbaseclasses/Makefile.am	2005-10-31 21:44:41.419408664 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@XBASEINCDIR@ 
 
 AM_CPPFLAGS=
diff -Nur b.hk_classes-0.8/hk_paradoxclasses/Makefile.am n.hk_classes-0.8/hk_paradoxclasses/Makefile.am
--- b.hk_classes-0.8/hk_paradoxclasses/Makefile.am	2005-05-05 15:56:52.000000000 +0200
+++ n.hk_classes-0.8/hk_paradoxclasses/Makefile.am	2005-11-03 18:39:22.632717584 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@PARADOXINCDIR@
 
 AM_CPPFLAGS=
diff -Nur b.hk_classes-0.8/hk_mdbclasses/Makefile.am n.hk_classes-0.8/hk_mdbclasses/Makefile.am
--- b.hk_classes-0.8/hk_mdbclasses/Makefile.am	2005-06-25 10:39:15.000000000 +0200
+++ n.hk_classes-0.8/hk_mdbclasses/Makefile.am	2005-11-03 18:39:22.594723360 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@MDBINCDIR@
 
 AM_CPPFLAGS=$(GLIB_CFLAGS)
