sed -i 's%libdir=@HK_CLASSESDIR@/drivers%libdir=@HK_DRIVERSDIR@%'

--- hk_classes-0.5.5/acinclude.m4.orig	Sun Aug 18 16:35:37 2002
+++ hk_classes-0.5.5/acinclude.m4	Thu Nov 21 11:47:35 2002
@@ -29,6 +29,15 @@
  HK_CLASSESDIR=$prefix
  HK_CLASSESINCDIR=$HK_CLASSESDIR/include
 fi
+if test "$with_hk_classes_dir" != ""; then
+HK_CLASSESDIR=$with_hk_classes_dir
+fi
+if test "$with_hk_classes_drvdir" != ""; then
+HK_DRIVERSDIR=$with_hk_classes_drvdir
+else
+HK_DRIVERSDIR=$HK_CLASSESDIR/drivers
+fi
+AC_SUBST(HK_DRIVERSDIR)
 if test "$with_hk_classes_incdir" != ""; then
 HK_CLASSESINCDIR=$with_hk_classes_incdir
 fi
--- hk_classes-0.5.5/configure.in.orig	Sat Sep 21 16:41:31 2002
+++ hk_classes-0.5.5/configure.in	Thu Nov 21 11:48:16 2002
@@ -8,6 +8,8 @@
 AC_ARG_WITH(mysql-dir,[  --with-mysql-dir=PATH   Path to the Mysql directory],:,:)
 AC_ARG_WITH(mysql-incdir,[  --with-mysql-incdir=PATH   Path to the Mysql header files],:,:)
 AC_ARG_WITH(mysql-libdir,[  --with-mysql-libdir=PATH   Path to the Mysql library],:,:)
+AC_ARG_WITH(hk_classes-dir,[  --with-hk_classes-dir=PATH   Path for the hk_classes library files],:,:)
+AC_ARG_WITH(hk_classes-drvdir,[  --with-hk_classes-drvdir=PATH   Path for the hk_classes driver files],:,:)
 AC_ARG_WITH(hk_classes-incdir,[  --with-hk_classes-incdir=PATH   Path for the hk_classes header files],:,:)
 
 AC_ARG_WITH(postgres-dir,[  --with-postgres-dir=PATH   Path to the postgres directory],:,:)
--- hk_classes-0.5.5/hk_mysqlclasses/Makefile.am.orig	Wed Sep  4 19:11:59 2002
+++ hk_classes-0.5.5/hk_mysqlclasses/Makefile.am	Thu Nov 21 12:44:43 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@MYSQLINCDIR@
 AM_CXXFLAGS= -Wall
 
--- hk_classes-0.5.5/hk_postgresclasses/Makefile.am.orig	Tue Aug 13 19:58:33 2002
+++ hk_classes-0.5.5/hk_postgresclasses/Makefile.am	Thu Nov 21 12:44:58 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES=   -I../hk_classes -I@POSTGRESINCDIR@
 
 libhk_postgresdriver_la_LDFLAGS = -version-info 0:0:0 -lpq  -L../hk_classes -L@POSTGRESLIBDIR@
--- hk_classes-0.5.5/hk_odbcclasses/Makefile.am.orig	Mon Nov  4 19:41:17 2002
+++ hk_classes-0.5.5/hk_odbcclasses/Makefile.am	Tue Dec  3 17:13:54 2002
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@ODBCINCDIR@ 
 AM_CXXFLAGS= -Wall
 
--- hk_classes-0.7.2/hk_sqliteclasses/Makefile.am.orig	2005-02-18 08:47:12.000000000 +0100
+++ hk_classes-0.7.2/hk_sqliteclasses/Makefile.am	2005-02-18 08:49:37.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@SQLITEINCDIR@
 
 AM_CXXFLAGS= -Wall
--- hk_classes-0.7.2/hk_sqlite3classes/Makefile.am.orig	2005-02-18 08:49:58.000000000 +0100
+++ hk_classes-0.7.2/hk_sqlite3classes/Makefile.am	2005-02-18 08:50:08.000000000 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@SQLITE3INCDIR@
 
 AM_CXXFLAGS= -Wall
--- hk_classes-0.7.4/hk_firebirdclasses/Makefile.am~	2005-07-09 22:06:23.154547992 +0200
+++ hk_classes-0.7.4/hk_firebirdclasses/Makefile.am	2005-07-09 22:05:32.071313824 +0200
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@FIREBIRDINCDIR@
 
 AM_CPPFLAGS=
diff -Nur b.hk_classes-0.8/hk_mdbclasses/Makefile.am n.hk_classes-0.8/hk_mdbclasses/Makefile.am
--- b.hk_classes-0.8/hk_mdbclasses/Makefile.am	2005-06-25 10:39:15.000000000 +0200
+++ n.hk_classes-0.8/hk_mdbclasses/Makefile.am	2005-11-03 13:39:22.594723360 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@MDBINCDIR@
 
 AM_CPPFLAGS=$(GLIB_CFLAGS)
diff -Nur b.hk_classes-0.8/hk_paradoxclasses/Makefile.am n.hk_classes-0.8/hk_paradoxclasses/Makefile.am
--- b.hk_classes-0.8/hk_paradoxclasses/Makefile.am	2005-05-05 15:56:52.000000000 +0200
+++ n.hk_classes-0.8/hk_paradoxclasses/Makefile.am	2005-11-03 13:39:22.632717584 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@PARADOXINCDIR@
 
 AM_CPPFLAGS=
diff -Nur b.hk_classes-0.8/hk_xbaseclasses/Makefile.am n.hk_classes-0.8/hk_xbaseclasses/Makefile.am
--- b.hk_classes-0.8/hk_xbaseclasses/Makefile.am	2005-04-10 11:55:26.000000000 +0200
+++ n.hk_classes-0.8/hk_xbaseclasses/Makefile.am	2005-11-03 13:39:22.700707248 +0100
@@ -1,4 +1,4 @@
-libdir=@HK_CLASSESDIR@/drivers
+libdir=@HK_DRIVERSDIR@
 INCLUDES= -I../hk_classes -I@XBASEINCDIR@ 
 
 AM_CPPFLAGS=
diff -Nur b.hk_classes-0.8/hk_classes/hk_drivermanager.cpp n.hk_classes-0.8/hk_classes/hk_drivermanager.cpp
--- b.hk_classes-0.8/hk_classes/hk_drivermanager.cpp	2005-09-17 21:07:01.000000000 +0200
+++ n.hk_classes-0.8/hk_classes/hk_drivermanager.cpp	2005-11-03 18:25:30.665093872 +0100
@@ -60,7 +60,7 @@
 };
 
 requestdialogtype* hk_drivermanager::p_driverselectdialog=NULL;
-hk_string hk_drivermanager::p_hk_classespath=(hk_string)HKCLASSES+"/drivers";
+hk_string hk_drivermanager::p_hk_classespath=(hk_string)HK_DRIVERSDIR;
 vector<hk_string> hk_drivermanager::p_driverlist;
 
 typedef hk_connection*  connectiontype(hk_drivermanager* );
@@ -83,8 +83,7 @@
 {
     p_private = new hk_drivermanagerprivate;
     hk_report::setup_reportbasics();
-    p_hk_classespath=HKCLASSES;
-    p_hk_classespath=p_hk_classespath+"/drivers";
+    p_hk_classespath=HK_DRIVERSDIR;
     scan_directory();
     char* h= getenv("HOME");
     hk_string classespath=(h==NULL?"/tmp":h);
@@ -449,7 +448,7 @@
 //the following is a hack for LFS
             if (b.find("/usr/local/hk_classes")<b.size())
             {
-                b=HKCLASSES;b+="/drivers";
+                b=HK_DRIVERSDIR;
 
                 set_path(b);
                 show_warningmessage("Due to a new directory hierarchy, your preferences have been changed");
diff -Nur b.hk_classes-0.8/hk_classes/hk_drivermanager.h n.hk_classes-0.8/hk_classes/hk_drivermanager.h
--- b.hk_classes-0.8/hk_classes/hk_drivermanager.h	2005-06-22 20:12:09.000000000 +0200
+++ n.hk_classes-0.8/hk_classes/hk_drivermanager.h	2005-11-03 18:16:15.266527288 +0100
@@ -17,6 +17,14 @@
 #include "hk_url.h"
 #include "hk_definitions.h"
 #include "hk_string.h"
+
+#ifndef HK_DRIVERSDIR
+#ifndef HKCLASSES
+#define HKCLASSES "/usr/local/lib/hk_classes"
+#endif
+#define HK_DRIVERSDIR HKCLASSES "/driver"
+#endif
+
 using namespace std;
 
 class hk_connection;
--- b.hk_classes-0.8/hk_classes/Makefile.am	2005-08-24 14:17:01.000000000 +0200
+++ n.hk_classes-0.8/hk_classes/Makefile.am	2005-11-03 19:45:52.286247208 +0100
@@ -25,7 +25,7 @@
 
 libhk_classes_la_LIBADD =$(LIBPYTHON)  $(PYTHONLIB)
 
-libhk_classes_la_CPPFLAGS= -DHKCLASSES=\"$(HK_CLASSESDIR)\"
+ 
 
 
 libhk_classes_la_SOURCES= \
--- b.hk_classes-0.8/acinclude.m4	2005-10-31 21:41:43.000000000 +0100
+++ n.hk_classes-0.8/acinclude.m4	2005-11-03 23:24:53.212345792 +0100
@@ -52,7 +52,8 @@
 AC_SUBST(HK_CLASSESINCDIR)
 NODRIVER=true
 POSSIBLE_DRIVERS="FIREBIRD,MYSQL,ODBC,PARADOX,POSTGRESQL,SQLITE2,SQLITE3,MDB,XBASE"
-dnl AC_DEFINE_UNQUOTED(HKCLASSES, "$HK_CLASSESDIR",[PREPROCESSOR DEFINITION])
+AC_DEFINE_UNQUOTED(HKCLASSES, "$HK_CLASSESDIR", [HKCLASSES - directory for libhk_classes library])
+AC_DEFINE_UNQUOTED(HK_DRIVERSDIR, "$HK_DRIVERSDIR", [HK_DRIVERSDIR - database drivers' directory])
 AC_MSG_RESULT(OK)
 ])
 
