--- hk_classes-0.8/acinclude.m4~	2005-11-03 18:00:12.015699640 +0100
+++ hk_classes-0.8/acinclude.m4	2005-11-03 18:00:44.578749304 +0100
@@ -903,8 +903,8 @@
 if test "$want_mdb" = "yes"; then
 AC_MSG_CHECKING([whether MDBTOOLS is available])
 if test "$with_mdb_dir" = ""; then
-   MDBLIBDIR=/usr/local/lib
-   MDBINCDIR=/usr/local/include
+   MDBLIBDIR=/usr/lib
+   MDBINCDIR=/usr/include/mdb
 else MDBLIBDIR=$with_mdb_dir/lib
 MDBINCDIR=$with_mdb_dir/include
 fi
@@ -926,7 +926,7 @@
 CPPFLAGS="$ac_save_CPPFLAGS -I .  -I $MDBINCDIR $GLIB_CFLAGS "
 
 AC_TRY_LINK([#include <mdbtools.h>],[
-mdb_init();
+void mdb_init();
 ;],[mdb_lib=yes],[mdb_lib=no])
 
 AC_MSG_RESULT($mdb_lib)
