--- acinclude.m4.orig	2005-02-16 10:16:24.694878688 +0100
+++ acinclude.m4	2005-02-16 10:18:01.651139096 +0100
@@ -661,10 +661,10 @@
   LIBPYTHON=-lpython$version
 fi
 
-python_libdirs="$ac_python_dir/lib /usr/lib /usr/local /usr/lib $kde_extra_libs"
-AC_FIND_FILE(python$version/copy.py, $python_libdirs, python_moddir)
+python_libdirs="$ac_python_dir/lib /usr/lib /usr/local /usr/lib /usr/share $kde_extra_libs"
+AC_FIND_FILE(python$version/copy.pyc, $python_libdirs, python_moddir)
 python_moddir=$python_moddir/python$version
-if test ! -r $python_moddir/copy.py; then
+if test ! -r $python_moddir/copy.pyc; then
   python_moddir=no
 fi
 
